Unprefix -webkit-box-decoration-break

karlcow opened the pull request:
#### b5ebfd11c568e682a57267ee4e38a9050d2663f0
<pre>
Unprefix -webkit-box-decoration-break
<a href="https://bugs.webkit.org/show_bug.cgi?id=229078">https://bugs.webkit.org/show_bug.cgi?id=229078</a>
<a href="https://rdar.apple.com/82176202">rdar://82176202</a>

Reviewed by NOBODY (OOPS!).

This is unprefixing -webkit-box-decoration-break. But it also keeps
-webkit-box-decoration-break as an alias, as there is a lot of usage
online with the prefixed version.
<a href="https://chromestatus.com/metrics/css/timeline/popularity/414">https://chromestatus.com/metrics/css/timeline/popularity/414</a>
Probably an additional work would be to remove the prefixed version
from WPT tests and extends the tests if necessary.
<a href="https://github.com/web-platform-tests/interop/issues/486">https://github.com/web-platform-tests/interop/issues/486</a>

A series of tests related to css-break needs to be solved.
<a href="https://github.com/WebKit/WebKit/pull/17007#issuecomment-2519257965">https://github.com/WebKit/WebKit/pull/17007#issuecomment-2519257965</a>

But in most cases, this change doesn&apos;t affect the current rendering
of the pages. A lot of web developers are already using the two
unprefixed/prefixed combination on their sites. There are already
expecting the current behavior of the prefixed version.
We can find some web pages with only the unprefixed version. In this
case, this patch will improve the websites layout.

The fragmentation on two columns doesn&apos;t affect the layout in any
significant ways.

You can see experimental renderings with the unprefixed version on
<a href="https://github.com/WebKit/WebKit/pull/17007#issuecomment-2520170733">https://github.com/WebKit/WebKit/pull/17007#issuecomment-2520170733</a>

* LayoutTests/fast/box-decoration-break/box-decoration-break-parsing-expected.txt:
* LayoutTests/fast/box-decoration-break/box-decoration-break-parsing.html:
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/fast/css/getComputedStyle/resources/property-names.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-break/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-break/parsing/box-decoration-break-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-break/parsing/box-decoration-break-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001-expected.txt:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Removed.
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Removed.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::conservativelyCollectChangedAnimatableProperties const):
* Source/WebInspectorUI/UserInterface/Models/CSSKeywordCompletions.js:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b5ebfd11c568e682a57267ee4e38a9050d2663f0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79508 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58517 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32885 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/84087 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30601 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/81642 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67610 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6796 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62178 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/20038 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82579 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/52232 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72438 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42488 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49580 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/26601 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/29017 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70699 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/27059 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85498 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6774 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/4715 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70427 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-view-transitions/snapshot-containing-block-includes-scrollbar-gutter.html (failure)") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6939 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68300 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69671 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13698 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12591 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/6726 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/12413 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6622 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/10104 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8421 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e1e04f60319d4d488642b6c94dd2079945577cdf. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/10527171ab2a12bba621a9aa1ab1c39130659a7c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

karlcow commented:
Once I get the tests right I need to adjust the commit message.

nt1m commented on LayoutTests/fast/box-decoration-break/box-decoration-break-parsing.html:
```diff
@@ -30,6 +30,19 @@
 shouldBeEqualToString("e.style.getPropertyValue('-webkit-box-decoration-break')", "");
 shouldBe("computedStyle.getPropertyValue('-webkit-box-decoration-break')", "'slice'");
 
+e.style.boxDecorationBreak = "slice";
+shouldBe("e.style.getPropertyValue('box-decoration-break')", "'slice'");
+shouldBe("computedStyle.getPropertyValue('box-decoration-break')", "'slice'");
+
+e.style.boxDecorationBreak = "clone";
+shouldBe("e.style.getPropertyValue('box-decoration-break')", "'clone'");
+shouldBe("computedStyle.getPropertyValue('box-decoration-break')", "'clone'");
+
+e.style.boxDecorationBreak = "";
+e.style.boxDecorationBreak = "invalid";
+shouldBeEqualToString("e.style.getPropertyValue('box-decoration-break')", "");
+shouldBe("computedStyle.getPropertyValue('box-decoration-break')", "'slice'");
+
```
WPT already covers the unprefixed parts, we don't really need this. Maybe we could add `test_invalid_value` to the WPT for the last case.

karlcow replied:
Once this bot run is done. I'll fix things. Thanks.

karlcow replied:
https://wpt.fyi/results/css/css-break?label=master&label=experimental&aligned&q=box-decoration-break

I could add to this one indeed.
https://github.com/web-platform-tests/wpt/blob/57c9cc80fb/css/css-break/parsing/box-decoration-break-invalid.html

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/5d1b74f20922bfd088b86281f86e98671e8a3f45. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

karlcow commented:
There is this comment too on WPT 
https://github.com/web-platform-tests/interop/issues/486#issuecomment-1732372123

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1fbf9aa1b41e5ba820b2012533cb6d907e78175b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/884248bb1008259931e721c8e18b4ec004f85ff5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/dc064e603a7383d3aed2bf8b112f3fc903c01510. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/adbcd0653461c2fde3d2cc3953d11cdbb217769f. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

karlcow commented:
I created a new issue for missing tests on WPT https://github.com/web-platform-tests/wpt/issues/43440

darinadler reviewed (approved):
Change log says this keeps `box-decoration-break` as an alias. You meant to say this keeps `-webkit-box-decoration-break` as an alias.

karlcow commented:
> Change log says this keeps `box-decoration-break` as an alias. You meant to say this keeps `-webkit-box-decoration-break` as an alias.

Thanks @darinadler I just push an update of the commit message.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/f2231f563cb52a88847d482cb1a7ee12f8ef6760. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

karlcow commented:
In https://bugs.chromium.org/p/chromium/issues/detail?id=682224
https://bugs.chromium.org/p/chromium/issues/detail?id=481431 box-decoration-break has no effect in multicol and paged media
And https://bugs.webkit.org/show_bug.cgi?id=144249

Let's reimport css/css-break

karlcow commented:
This is being done in https://github.com/WebKit/WebKit/pull/21138

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/da3f88e085215589fafe5157e9d92a01a19e6cef. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/da3f88e085215589fafe5157e9d92a01a19e6cef. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3da0a9d26698bb61f64e2d1b541a3c26194376c8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3da0a9d26698bb61f64e2d1b541a3c26194376c8. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

karlcow commented:
Related bug https://bugs.webkit.org/show_bug.cgi?id=145808
Related test https://www.software.hixie.ch/utilities/js/live-dom-viewer/?saved=12030

karlcow commented:
This has not landed because WebKit needs support for support for paginated/columnated fragmentation.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/f4bb3da93eef0f51e58e95c2007ed71f0f0d44bd. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

karlcow commented:
Most of the code on GitHub is using either the unprefixed version, or the two versions prefixed/unprefixed.

karlcow commented:
Rebasing to the current version of the code.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/6ebfe100767945a44137655af2ac29a1a66c7f34. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

smfr commented:
> This is unprefixing -webkit-box-decoration-break. But it also keeps
-webkit-box-decoration-break as an alias, as there is a lot of usage
online with the prefixed version.

I want to see some words here about what's still broken and why it's OK to unprefix.

karlcow commented:
Pushing first a PR to see what is failing, Then I will amend the commit message for things which are still breaking.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/22f9d68c39559db1d45227ead36a5ad835f8198a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

karlcow commented:
This will not fix for Safari only

https://wpt.fyi/results/css/css-break/box-decoration-break-clone-001.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-002.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-004.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-006.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-008.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-009.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-010.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-012.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-014.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-015.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-016.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-017.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-018.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-019.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-021.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-022.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-023.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-024.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-025.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-026.html

These are failing in Safari and Firefox, and they will not be fixed either. 

https://wpt.fyi/results/css/css-break/box-decoration-break-clone-020.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-027.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-028.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-029.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-030.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-031.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-033.html
https://wpt.fyi/results/css/css-break/box-decoration-break-clone-034.html

karlcow commented:
OK this should all be green. There were infra failures today. 
I will amend the commit. and Will make another run.

karlcow commented:
From left to right. Rendering of text with only unprefixed box-decoration-break

* STP 208
* MiniBrowser with unprefixed 
* Firefox 
* Chrome

![Screenshot 2024-12-05 at 21 17 59](https://github.com/user-attachments/assets/5605c17d-5d07-4a93-99d0-7857197fe606)

In the simple case we can see that it doesn't create any issues. Quite the opposite for people who would not have added the prefix in their site.

karlcow commented:
And this is a similar text with the text fragmented on two columns. 

![Screenshot 2024-12-05 at 21 24 02](https://github.com/user-attachments/assets/e627f160-f2b9-4384-abf8-b0d046c65489)

Unprefixing doesn't create major issues for the most common cases.

karlcow committed:
Unprefix -webkit-box-decoration-break
https://bugs.webkit.org/show_bug.cgi?id=229078
rdar://82176202

Reviewed by NOBODY (OOPS!).

This is unprefixing -webkit-box-decoration-break. But it also keeps
-webkit-box-decoration-break as an alias, as there is a lot of usage
online with the prefixed version.
https://chromestatus.com/metrics/css/timeline/popularity/414
Probably an additional work would be to remove the prefixed version
from WPT tests and extends the tests if necessary.
https://github.com/web-platform-tests/interop/issues/486

A series of tests related to css-break needs to be solved.
https://github.com/WebKit/WebKit/pull/17007#issuecomment-2519257965

But in most cases, this change doesn't affect the current rendering
of the pages. A lot of web developers are already using the two
unprefixed/prefixed combination on their sites. There are already
expecting the current behavior of the prefixed version.
We can find some web pages with only the unprefixed version. In this
case, this patch will improve the websites layout.

The fragmentation on two columns doesn't affect the layout in any
significant ways.

You can see experimental renderings with the unprefixed version on
https://github.com/WebKit/WebKit/pull/17007#issuecomment-2520170733

* LayoutTests/fast/box-decoration-break/box-decoration-break-parsing-expected.txt:
* LayoutTests/fast/box-decoration-break/box-decoration-break-parsing.html:
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt:
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt:
* LayoutTests/fast/css/getComputedStyle/resources/property-names.js:
* LayoutTests/imported/w3c/web-platform-tests/css/css-break/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-break/parsing/box-decoration-break-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-break/parsing/box-decoration-break-valid-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-001-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-001-expected.txt:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Removed.
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt: Removed.
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::conservativelyCollectChangedAnimatableProperties const):
* Source/WebInspectorUI/UserInterface/Models/CSSKeywordCompletions.js:

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b5ebfd11c568e682a57267ee4e38a9050d2663f0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/17007

karlcow commented:
Adding the rendering with a block element. being split on two columns

<img width="2038" alt="Screenshot 2024-12-06 at 07 13 30" src="https://github.com/user-attachments/assets/cd8b9d36-9709-40a1-84e8-783474988d70">

karlcow commented:
https://blog.kizu.dev/box-decoration-break/

karlcow commented:
Closing the pull request.
